google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/JsonStreamWriterTest.java

```diff
@@ -581,4 +582,25 @@ public void run() throws Throwable {
                   "Exceeds client side inflight buffer, consider add more buffer or open more connections"));
     }
   }
+
+  @Test
+  public void testFlowControlSettingNoLimitBehavior() throws Exception {
```

Comment:
Discussed offline, from this test and the previous test I'm thinking the current behavior is not so ideal, in the throw exception situation, since we may successfully send out the request, however we will still throw exception
the client might think they need to resend the request, however they don't need.

Reply:
good catch, d.